pkg/scorecard_result.go

```diff
@@ -38,7 +39,7 @@ type ScorecardResult struct {
 
 // AsJSON outputs the result in JSON format with a newline at the end.
 // If called on []ScorecardResult will create NDJson formatted output.
-func (r *ScorecardResult) AsJSON(showDetails bool, writer io.Writer) error {
+func (r *ScorecardResult) AsJSON(showDetails bool, logLevel zapcore.Level, writer io.Writer) error {
```

Comment:
Can we use the `DetailType` enum defined in this PR instead of the `zapcore` package?

It would be nice to simplify the code by removing third-party dependencies like `zap` unless they are providing some advantage over native Golang logging packages.

Reply:
I don't disagree, but let's do that in a follow-up CL to cleanup level/detail level (I don't have full context on the need for zap package). scorecard currently takes a `--verbosity` argument that needs to be changed as well.
Created https://github.com/ossf/scorecard/issues/717 for tracking the issue.

Reply:
SGTM